### ISSUE TITLE
Implement training

### ISSUE THREAD

**Issue by RecurvedBow** [RecurvedBow/image-synthesis-diffusion-model#7]
Implement the loss function and the training. Use (https://arxiv.org/pdf/2006.11239.pdf; 3.4) as loss function:
![image](https://user-images.githubusercontent.com/101172476/186731059-3c8f94d6-3808-4e69-832a-210ecd839b15.png)

Note: Expected noise is a sample of the normal distribution; input of network is the noisy image given alpha (product variance) & timestep t.
